src/project.coffee

```diff
@@ -311,6 +309,7 @@ class Project extends Model
 
   buildEditorForBuffer: (buffer, editorOptions) ->
     editor = new TextEditor(_.extend({buffer, registerEditor: true}, editorOptions))
+    editor.setEncoding(atom.config.get(editor.getRootScopeDescriptor(), "core.fileEncoding"))
```

Comment:
This should probably go in the `TextEditor` constructor now so that no matter how it is created, it is setting the encoding from the root scope descriptor.